velox/functions/prestosql/window/LeadLag.cpp

```diff
@@ -141,8 +141,13 @@ class LeadLagFunction : public exec::WindowFunction {
       if (offsets_->isNullAt(i)) {
         rowNumbers_[i] = kNullRow;
       } else {
-        vector_size_t offset = offsets_->valueAt(i);
+        auto offset = offsets_->valueAt(i);
         VELOX_USER_CHECK_GE(offset, 0, "Offset must be at least 0");
+        // Set rowNumber to kNullRow for out of range offset.
+        if (offset > partition_->numRows()) {
```

Comment:
Is this a fix for another bug not related to offset being > max 32-bit integer? How did we handle offsets with 32-bit integer range, but larger than partition_->numRows() before this change?

Reply:
It is still fixing the same issue: before this fix, the larger than int32 offset can be passed to function: `rowNumberIgnoreNull`, its signature: 

```c++
  vector_size_t rowNumberIgnoreNull(
      const uint64_t* rawNulls,
      vector_size_t offset,
      vector_size_t start,
      vector_size_t end,
      vector_size_t step)
```

The offset is of type `vector_size_t`, if the offset is the large offset like `(int64_t) std::numeric_limits<uint32_t>::max() + 2`, the real offset rowNumberIgnoreNull sees is actually `1`(casted to int32).

So this `if` pre-intercept out of range offset.

Reply:
Got it. Is there similar problem in setRowNumbersForConstantOffset ?

Reply:
Yeah, good catch, has modified accordingly.

Reply:
@mbasmanova : Good catch !